src/persist-client/src/read.rs

```diff
@@ -62,10 +64,10 @@ impl ReaderId {
 ///
 /// See [ReadHandle::snapshot] for details.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct SnapshotSplit {
+pub struct SnapshotSplit<T> {
     shard_id: ShardId,
     as_of: Vec<[u8; 8]>,
-    batches: Vec<String>,
+    batches: Vec<(String, Description<T>)>,
```

Comment:
SnapshotSplit needs to be something we can send over the network, so you'll have to encode the Description into a DescriptionMeta like we do in StateRollupMeta, instead of adding a <T> to SnapshotSplit. then you can rehydrate the Description<T> in ReadHandle::snapshot_iter